drivers/media/i2c/irs1125.c

```diff
@@ -0,0 +1,1137 @@
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_ILLU_ON,
+        .name = "Turn illu on or off",
+        .type = V4L2_CTRL_TYPE_BOOLEAN,
+        .flags = V4L2_CTRL_FLAG_EXECUTE_ON_WRITE,
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 1
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT0,
+        .name = "Get ident 0 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT1,
+        .name = "Get ident 1 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT2,
+        .name = "Get ident 2 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+};
+
+static int irs1125_detect(struct v4l2_subdev *sd)
+{
+    u16 read;
+    int ret;
+    struct i2c_client *client = v4l2_get_subdevdata(sd);
+
+    ret = irs1125_read(sd, IRS1125_REG_DESIGN_STEP, &read);
+    if (ret < 0)
+    {
+        dev_err(&client->dev, "error reading from i2c\n");
+        return ret;
+    }
+
+    if (read != 0x0a12) {
+        dev_err(&client->dev, "Design step expected 0x0a12 got %x", read);
+        return -ENODEV;
+    }
+
+    return 0;
+}
+
+static int irs1125_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
+{
+    struct v4l2_mbus_framefmt *format =
+                v4l2_subdev_get_try_format(sd, fh->pad, 0);
+    struct v4l2_rect *crop =
+                v4l2_subdev_get_try_crop(sd, fh->pad, 0);
+
+    crop->left = IRS1125_COLUMN_START_DEF;
+    crop->top = IRS1125_ROW_START_DEF;
+    crop->width = IRS1125_WINDOW_WIDTH_DEF;
+    crop->height = IRS1125_WINDOW_HEIGHT_DEF;
+
+    format->code = MEDIA_BUS_FMT_Y12_1X12;
+
+    format->width = IRS1125_WINDOW_WIDTH_DEF;
+    format->height = IRS1125_WINDOW_HEIGHT_DEF;
+    format->field = V4L2_FIELD_NONE;
+    format->colorspace = V4L2_COLORSPACE_RAW;
+
+    return 0;
+}
+
+static const struct v4l2_subdev_internal_ops irs1125_subdev_internal_ops = {
+    .open = irs1125_open,
+};
+
+static int irs1125_parse_dt(struct device_node *np, struct irs1125 *sensor)
+{
+    struct v4l2_fwnode_endpoint bus_cfg;
+    struct device_node *ep;
+
+    int ret;
+
+    ep = of_graph_get_next_endpoint(np, NULL);
+    if (!ep)
+        return -EINVAL;
+
+    ret = v4l2_fwnode_endpoint_parse(of_fwnode_handle(ep), &bus_cfg);
+
+    if (!ret)
+        sensor->flags = bus_cfg.bus.mipi_csi2.flags;
```

Comment:
sensor->flags appears to be never used, so remove it. I suspect that means you can probably remove the whole of this function.

Having said that, you should probably check the number of CSI2 lanes configured in DT and ensure that it matches your register set.

Reply:
Removed, I also added proper configuration of CSI2 data lanes